### ISSUE TITLE
Is it possible to overwrite the font?

### ISSUE THREAD

**Issue by AntoniosBarotsis** [pawroman/zola-theme-terminimal#52]
I have currently added the following in my own `scss` file:

```scss
* {
  font-family: '...' !important;
}
```

Which works fine, *but*, the `hack` font is still downloaded, wasting some amount of time on the initial page load. Is that something I could change without making changes to the theme's code itself?

Edit: *I realize that this might be more of a zola question than a theme question but I couldn't find anything in the docs so I'm just asking here for now*.

**Comment by marcospb19** [comment]
I believe you have to change the theme code itself in order to change that.

**Comment by pawroman** [comment]
It's not that hard to change this, the theme would have to basically expose an extra switch.

I would be glad to accept this as a contribution, e.g. a true by default switch to enable the Hack font (e.g. `config.extra.use_hack_font`).

See https://github.com/pawroman/zola-theme-terminimal/blob/master/templates/macros/head.html#L22-L26

**Comment by AntoniosBarotsis** [comment]
So I tried commenting out lines 22-26 completely but running `zola build` still puts all the fonts in my newly created public folder. That shouldn't be happening right 🤔

**Comment by AntoniosBarotsis** [comment]
This would *probably* work fine if we completely removed the fonts from the `static` folder and imported them like I do [here](https://github.com/AntoniosBarotsis/antoniosbarotsis.github.io/blob/1c5ea87107e10de8a20fe1a4e1a52b04fd4bdee5/templates/index.html#L18-L19) (in retrospect, I am not even sure if this is even needed, anything UI related is not my strongest area 😅) and add the remote urls in a `scss` file (like [here](https://github.com/AntoniosBarotsis/antoniosbarotsis.github.io/blob/7a7702942412489386a10eb30077ca60db1e21fb/sass/index.scss#L55-L67)) instead of pointing them to the `fonts/` folder.

The `scss` file would work the same way you are now choosing between the hack font subset or opting in to the full one [here](https://github.com/pawroman/zola-theme-terminimal/blob/39d13a9a6a1b7b6aaaa3e567ccdb978d8d0ceced/templates/macros/head.html#L22-L26).

It could look something like this:

```html
{%- if config.extra.use_full_hack_font %}
    <link rel="stylesheet" href="{{ get_url(path="font-hack.css", trailing_slash=false) | safe }}">
{% else if config.extra.remove_hack_font %}
{% else %}
    <link rel="stylesheet" href="{{ get_url(path="font-hack-subset.css", trailing_slash=false) | safe }}">
{% endif -%}
```

I think having the hack font be opt-out is better so as to not break everyone's blogs with the new update; you would have to change your config only to *remove* the font, otherwise everything should stay the same, hence the `remove_hack_font`.

Let me know your thoughts!

**Comment by AntoniosBarotsis** [comment]
An update on this, I made the changes in my fork [here](https://github.com/pawroman/zola-theme-terminimal/compare/master...AntoniosBarotsis:zola-theme-terminimal:remove_hack_font?expand=1) and used it in my blog.

Without changing anything:

1. The `static/fonts` folder was no longer there
2. `font-hack-subset.css` was included (checked this via the browser `sources` tab

Setting `remove_hack_font = true` in my blog's `config.toml` removes the `font-hack-subset.css` file.

Is this something that would work for you as well? Maybe it's a good idea to check manually [this time](https://github.com/pawroman/zola-theme-terminimal/commit/39d13a9a6a1b7b6aaaa3e567ccdb978d8d0ceced#commitcomment-135136633) 😅, I've modified a bunch of files on my blog and there's some inconsistencies sometimes with the original.

**Comment by pawroman** [comment]
@goingforbrooke was working on adding a new/custom font. Maybe you could work on a "proper" way forward here together?

I wouldn't oppose adding support for custom fonts to the theme.

**Comment by pawroman** [comment]
(See #69)

**Comment by AntoniosBarotsis** [comment]
Ok so, after spending the better part of the last half an hour trying to remember what I was even doing when making those changes, it turns out none of that was needed 😅 

I have overwritten `index.scss` with the following:

```scss
@font-face {
  font-family: 'Fira Code';
  font-style:  normal;
  font-weight: 400;
  src: url("https://cdnjs.cloudflare.com/ajax/libs/firacode/6.2.0/ttf/FiraCode-Regular.ttf") format("woff");
}

@font-face {
  font-family: 'Fira Code';
  font-style:  normal;
  font-weight: 800;
  src: url("https://cdnjs.cloudflare.com/ajax/libs/firacode/6.2.0/ttf/FiraCode-Bold.ttf") format("woff");
}


* {
  font-family: 'Fira Code' !important;
}
```

That is all you need to overwrite the font apparently. I am not sure what the issue was the first time I tried but after checking the network tab, it looks like the only font that gets downloaded is `Fira Code`, I guess the browser figures out that since I overwrite the `font-family` for everything, the hack font is not needed. So not sure where to go with this from here 😅 

I guess we could mention this somewhere that in case you want to change the font you can do it this way?

**Comment by goingforbrooke** [comment]
As @pawroman mentioned, I added Exo2 as the font for [my blog](https://www.goingforbrooke.com/personal_blog/), but I kept Hack as the monospace font.

I don't plan to PR it, but it involved hijacking [`use_hack_font` in `config.toml`](https://github.com/goingforbrooke/zola-theme-terminimal/blob/7b55d0498f8201dce5c2140118399bb6a8de1814/config.toml#L80) to trigger a (new) [`if` in `head.html`](https://github.com/goingforbrooke/zola-theme-terminimal/blob/7b55d0498f8201dce5c2140118399bb6a8de1814/templates/macros/head.html#L22-L23C17).

**Comment by AntoniosBarotsis** [comment]
@goingforbrooke You still download the hack font 😅 
![image](https://github.com/user-attachments/assets/4919daad-31be-45c1-a84e-ac8b730d9543)

That's probably the issue I was having, you need this I think: 

```scss
* {
  font-family: 'Fira Code' !important;
}
```

**Comment by goingforbrooke** [comment]
yup, @AntoniosBarotsis bc I want it for code chunks and stuff 👆🏻
